analyzers/src/SonarAnalyzer.Common/Helpers/CountComparisonSyntax.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Linq;
+
+namespace SonarAnalyzer.Helpers
+{
+    public readonly struct CountComparisonSyntax : IEquatable<CountComparisonSyntax>
+    {
+        public enum Kind
+        {
+            Equals,
+            NotEquals,
+            LessThan,
+            LessThanOrEqual,
+            GreaterThan,
+            GreaterThanOrEqual,
+        }
+
+        public CountComparisonSyntax(int? left, Kind @operator, int? right)
+        {
+            Left = left;
+            Operator = @operator;
+            Right = right;
+        }
+        public int? Left { get; }
+        public int? Right { get; }
+        public Kind Operator { get; }
+
+        public bool ExpressionIsLeft() => !Left.HasValue;
+        public bool IsValid() => Left.HasValue ^ Right.HasValue;
+        public bool IsEmpty() => Empties.Contains(Normalized());
+        public bool IsAny() => Anys.Contains(Normalized());
+        public bool IsEmptyOrAny() => IsEmpty() || IsAny();
```

Comment:
These are not trivial so they should be formatted like this (according to our latest coding style update)
```
public bool ExpressionIsLeft() => 
	!Left.HasValue;

```